zk_toolbox/crates/zk_inception/src/commands/update.rs

```diff
@@ -0,0 +1,202 @@
+        &shell.read_file(ecosystem.get_default_configs_path().join(CONTRACTS_FILE))?,
+    )?;
+    let secrets_path = ecosystem.get_default_configs_path().join(SECRETS_FILE);
+    let secrets = serde_yaml::from_str(&shell.read_file(secrets_path.clone())?)?;
+
+    for chain in ecosystem.list_of_chains() {
+        logger::step(msg_updating_chain(&chain));
+        let chain = ecosystem
+            .load_chain(Some(chain))
+            .context(MSG_CHAIN_NOT_FOUND_ERR)?;
+        update_chain(
+            shell,
+            &chain,
+            &general_config,
+            &external_node_config,
+            &genesis_config,
+            &contracts_config,
+            &secrets,
+            &secrets_path,
+        )?;
+    }
+
+    logger::outro(MSG_ZKSYNC_UPDATED);
+
+    Ok(())
+}
+
+fn merge_yaml_internal(
+    a: &mut serde_yaml::Value,
+    b: serde_yaml::Value,
+    current_key: serde_yaml::Value,
+    diff: &mut ConfigDiff,
+) -> anyhow::Result<()> {
+    match (a, b) {
+        (serde_yaml::Value::Mapping(a), serde_yaml::Value::Mapping(b)) => {
+            for (key, value) in b {
+                if a.contains_key(&key) {
+                    merge_yaml_internal(a.get_mut(&key).unwrap(), value, key, diff)?;
+                } else {
+                    a.insert(key.clone(), value.clone());
+                    diff.added_fields.insert(key, value);
+                }
+            }
+        }
+        (a, b) => {
+            if a != &b {
+                diff.value_diff.insert(current_key, b);
+            }
+        }
+    }
+    Ok(())
+}
+
+fn merge_yaml(a: &mut serde_yaml::Value, b: serde_yaml::Value) -> anyhow::Result<ConfigDiff> {
+    let mut diff = ConfigDiff::default();
+    merge_yaml_internal(a, b, "".into(), &mut diff)?;
+    Ok(diff)
+}
+
+fn save_updated_config(
+    shell: &Shell,
+    config: serde_yaml::Value,
+    path: &Path,
+    diff: ConfigDiff,
+    msg: &str,
+) -> anyhow::Result<()> {
+    if diff.added_fields.is_empty() {
+        return Ok(());
+    }
+
+    logger::info(msg);
+    logger::info(logger::object_to_string(&diff.added_fields));
+
+    let general_config = serde_yaml::to_string(&config)?;
+    shell.write_file(path, general_config)?;
+
+    Ok(())
+}
+
+fn update_chain(
+    shell: &Shell,
+    chain: &ChainConfig,
+    general: &serde_yaml::Value,
+    external_node: &serde_yaml::Value,
+    genesis: &serde_yaml::Value,
+    contracts: &serde_yaml::Value,
+    secrets: &serde_yaml::Value,
+    secrets_path: &Path,
+) -> anyhow::Result<()> {
+    let current_general_config_path = chain.path_to_general_config();
+    let mut current_general_config =
+        serde_yaml::from_str(&shell.read_file(current_general_config_path.clone())?)?;
+    let diff = merge_yaml(&mut current_general_config, general.clone())?;
+    save_updated_config(
+        shell,
+        current_general_config,
+        &current_general_config_path,
+        diff,
+        MSG_DIFF_GENERAL_CONFIG,
+    )?;
```

Comment:
Looks, like we need the function
`fn update_config(shell: Shell, original_config_path: Path, chain_config_path: Path, replace_config: bool, msg: &str)`